### PR TITLE
feat(chat): edit messages

### LINKED ISSUE #4723: chat - implement message edit
update your own state and announce to peers

### PR DESCRIPTION
**Pull request by molimauro**
### What this PR does 📖
Implement message edit functionality.

### Which issue(s) this PR fixes 🔨
- Resolve #4723

### Special notes for reviewers 🗒️
If you send a new message and edit it while the other user is offline, the latter will receive the original copy. (same problem of the reactions while offline)

### Additional comments 🎤
Noticing some problems with the enhancers, disabled here, will fix on a separate PR.

### CODE REVIEW

File: libraries/Iridium/chat/ChatManager.ts [hunk 1]
```diff
@@ -296,13 +297,22 @@ export default class ChatManager extends Emitter<ConversationMessage> {
         [fromDID]: reaction.reactions,
       }
 
-      this.state.conversations[conversationId] = {
-        ...this.state.conversations[conversationId],
-        message: {
-          ...this.state.conversations[conversationId].message,
-          [message.id]: message,
-        },
-      }
```
Comment (josephmcg) [inline libraries/Iridium/chat/ChatManager.ts:305]: does removing this cause reactivity issues? I'm not sure how Vue would detect the new object property

**Comment by WanderingHogan** [comment]
this is so sweet, i think we should hold off until we figure out the offline piece of it though since we are so close to EA, what do you all think?

File: libraries/Iridium/chat/ChatManager.ts (hunk 1 above)
Comment (molimauro) [reply libraries/Iridium/chat/ChatManager.ts:305]: It should not since the message obj is already there so it's already being tracked

**Comment by molimauro** [comment]
> this is so sweet, i think we should hold off until we figure out the offline piece of it though since we are so close to EA, what do you all think?

No problem of course ;)

File: libraries/Iridium/chat/ChatManager.ts (hunk 1 above)
Comment (josephmcg) [reply libraries/Iridium/chat/ChatManager.ts:305]: I thought this was an alternative way of writing
```
Vue.set(this.state.conversations[conversationId].message, [message.id], message)
```

Comment (josephmcg) [reply libraries/Iridium/chat/ChatManager.ts:305]: oh yeah I see. I thought this was concerning replies, not reacts. Should be alright then

Comment (molimauro) [reply libraries/Iridium/chat/ChatManager.ts:305]: Is replies reactivity working? Yesterday noticed some strange behaviour

Comment (josephmcg) [reply libraries/Iridium/chat/ChatManager.ts:305]: You're right, there are some issues on the sender side. I'll make a ticket and check it out tomorrow (unless you want to get it today)

Comment (molimauro) [reply libraries/Iridium/chat/ChatManager.ts:305]: Looks good, im trying to dig into this offline problem. Go with Vue.set :P

**Comment by phillsatellite** [comment]
Tested and showed up edited on both sides. Had the refresh bug the first time but after re adding friends and sending edited messages I could refresh and messages would still show up edited 🔥

**Review by WanderingHogan — APPROVED** [review]

**Comment by WanderingHogan** [comment]
@stavares843 @phillsatellite can you check this and merge if okay, otherwise comment here with screenshots

please check across all browsers, mac/win

**Review by stavares843 — APPROVED** [review]

**Pull request merged**